fdbserver/DataDistribution.actor.cpp

```diff
@@ -853,11 +833,6 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 			// We will get stuck at this! This only happens when a DC fails. No need to consider it right now.
 			if(!bestOption.present() && self->zeroHealthyTeams->get()) {
 				//Attempt to find the unhealthy source server team and return it
-				std::set<UID> completeSources;
-				for( int i = 0; i < req.completeSources.size(); i++ ) {
-					completeSources.insert( req.completeSources[i] );
-				}
-
 				int bestSize = 0;
 				for( int i = 0; i < req.completeSources.size(); i++ ) {
```

Comment:
Not sure how often duplicate servers exist in `req.completeSources`. 
If it is not rare, using the `complementSources` set calculated at the beginning of this function has performance benefit, since the `getTeam()` function is in hot path.

Reply:
Complete sources should never have duplicates, it was only turned into a set to do a count to lookup if ids exists. getTeam only called roughly once per shard moved, so it is not hot enough to the point that it should cause CPU problems for the DD algorithm.